qiskit/transpiler/target.py

```diff
@@ -0,0 +1,409 @@
+        if instruction_name in self._gate_map:
+            raise AttributeError("already in map")
+        self._gate_name_map[instruction_name] = instruction
+        qargs_val = {}
+        for qarg in qargs:
+            self.num_qubits = max(self.num_qubits, max(qarg) + 1)
+            if properties:
+                if qarg in properties:
+                    qargs_val[qarg] = properties[qarg]
+                else:
+                    qargs_val[qarg] = None
+            else:
+                qargs_val[qarg] = None
+            if qarg in self._qarg_gate_map:
+                self._qarg_gate_map[qarg].add(instruction_name)
+            else:
+                self._qarg_gate_map[qarg] = set(instruction_name)
+        self._gate_map[instruction_name] = qargs_val
+        self._coupling_graph = None
+        self._unweighted_dist_matrix = None
+        self._length_distance_matrix = None
+        self._error_distance_matrix = None
+        self._instruction_durations = None
+
+    @property
+    def qargs(self):
+        """The set of qargs in the gate map."""
+        return set(self._qarg_gate_map)
+
+    def get_qargs(self, gate):
+        """Get the qargs for a given gate instance
+
+        Args:
+           gate (str): The gate instance to get qargs for
+        Returns:
+            set: The set of qargs the gate instance applies to
+        """
+        return set(self._gate_map[gate])
+
+    def durations(self):
+        """Get an InstructionDurations object from the target
+
+        Returns:
+            InstructionDurations: The instruction duration represented in the
+                target
+        """
+        if self._instruction_durations is not None:
+            return self._instruction_durations
+        out_durations = []
+        for instruction, props_map in self._gate_map.items():
+            for qarg, properties in props_map.items():
+                if properties is not None and properties.length is not None:
+                    out_durations.append((instruction, list(qarg), properties.length, "s"))
+        self._instruction_durations = InstructionDurations(out_durations)
+        return self._instruction_durations
+
+    def get_gate_from_name(self, gate):
+        """Get the gate object for a given name
+
+        Args:
+           gate (str): The gate name to get the gate instance for
+        Returns:
+            Gate: The gate instance corresponding to the name
+        """
+        return self._gate_name_map[gate]
+
+    def get_qargs_from_name(self, gate_name):
+        """Get the qargs for a given gate by name
+
+        Args:
+           gate_name (str): The gate name to get qargs for. Note that gates
+            are not necessarily unique by name (ie different parameters, user
+            override, etc) this will return the qargs for any gate object that
+            has a name that mateches
+        Returns:
+            set: The set of qargs the gate applies to.
+        """
```

Comment:
So a method to go from a qarg to a list of gates. I assume in this case you mean the instances of the gate objects and not the names? Or would just the name be enough?

Reply:
Right, I would assume the user would want the full instances (since they're strictly more informative, and should only be marginally more costly to fetch than the names).

Reply:
Done in: https://github.com/Qiskit/qiskit-terra/pull/5885/commits/d10fce1b423a5ecc25f9ee60858325d52273ee96